### ISSUE TITLE
Issues UnIdentified By HTML Code Sniffer 

### ISSUE THREAD

**Issue by rakeshkmr516** [squizlabs/HTML_CodeSniffer#232]
Hi Team,

While performing Test, Observed below issues are unidentified by HTML code sniffer and AXE able to detect them ( Just would like to make sure we need to catch most of them) Any Reason?

WCAG 2.0 (A): MUST
WCAG Success Criteria:
1.3.1 Info and Relationships

```
> 1. <ul> and <ol> must only directly contain <li>, <script> or <template> elements
> 2. <li> elements must be contained in a <ul> or <ol> ( List item does not have a <ul>, <ol> or role="list" parent element)
> 3. Ensures elements with an ARIA role that require child roles contain them
```

WCAG 2.0 (A): MUST
Section 508: MUST
WCAG Success Criteria:
2.4.1 Bypass Blocks
Section 508 Guidelines:
1194.22 (o) Repetitive navigation link skip method

`> 1. The page does not contain a heading, skip link , or landmark region`

**Comment by ironikart** [comment]
Hi @rakeshkmr516 

I'm on the fence about the first issue. It's not strictly part of the WCAG success criteria as I understand it, but rather HTML validation. The validity of the HTML has a big impact on accessibility, obviously, but there are many tools that already validate this. I'll need to investigate it further.

The second issue is reported in HTMLCS as a 'Notice' due to it's difficulty to detect compliance programatically. The notice you will see is "Ensure that any common navigation elements can be bypassed; for instance, by use of skip links, header elements, or ARIA landmark roles.". Due to the variety of techniques you can use to achieve compliance it's best left to a human to confirm.

**Comment by rakeshkmr516** [comment]
Awesome. Thank you for response.
Hope we can get some interesting validation for the 1st issue

**Comment by ironikart** [comment]
Hi @rakeshkmr516. Regarding the first issue we aren't going to add the same failure that the AXE tool uses. The reason for this is that we don't validate HTML, only compliance to WCAG and the line between them in this case is blurred. My suggestion would be that HTMLCS should always be used in conjunction with a HTML validator that will detect these issues (based on doctype).
